pkg/server/plugin/nodeattestor/k8s/psat/psat.go

```diff
@@ -0,0 +1,269 @@
+package psat
+
+import (
+	"context"
+	"crypto"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/hcl"
+	"github.com/spiffe/spire/pkg/common/plugin/k8s"
+	"github.com/spiffe/spire/pkg/common/plugin/k8s/client"
+	"github.com/spiffe/spire/proto/common"
+	spi "github.com/spiffe/spire/proto/common/plugin"
+	"github.com/spiffe/spire/proto/server/nodeattestor"
+	"github.com/zeebo/errs"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+const (
+	pluginName = "k8s_psat"
+)
+
+var (
+	psatError                     = errs.Class("k8s-psat")
+	_         nodeattestor.Plugin = (*AttestorPlugin)(nil)
+)
+
+// NewAttestorPlugin creates a new PSAT node attestor plugin
+func NewAttestorPlugin() *AttestorPlugin {
+	return &AttestorPlugin{}
+}
+
+//AttestorPlugin is a PSAT (Projected SAT) node attestor plugin
+type AttestorPlugin struct {
+	mu     sync.RWMutex
+	config *attestorConfig
+}
+
+// ClusterConfig holds a single cluster configuration
+type ClusterConfig struct {
+	// API server public key file path.
+	// Public key is used for token validation
+	APIServerKeyFile string `hcl:"api_server_key_file"`
```

Comment:
curious why we changed this from ServiceAccountKeyFile? While it's true that the Kubernetes will use the API server key to sign SAT's if one isn't specified, it allows you to set a specific SAT key file (via `--service-account-key-file`).

Reply:
Got it. I didn't realize that other keys could be used. I'll change it back to `ServiceAccountKeyFile` :+1: